### PR TITLE
golangci: Enable revive linter

### PR DESCRIPTION
**Pull request by orestisfl**
### Summary of your changes
Enable revive linter and manually disable most rules that fail except some of the most obvious ones that we should fix and prevent.

This enables all rules that don't fail to prevent future bugs.

### Screenshot/Data
Fixes the following:

```text
launcher/launcher_test.go:300:16: datarace: datarace: range value tt is captured (by-reference) in goroutine (revive)
				time.Sleep(tt.delay)
				           ^
resources/providers/awslib/multi_region_test.go:27:2: duplicated-imports: Package "github.com/aws/aws-sdk-go-v2/aws" already imported (revive)
	awssdk "github.com/aws/aws-sdk-go-v2/aws"
	^
resources/providers/awslib/multi_region.go:28:2: duplicated-imports: Package "github.com/aws/aws-sdk-go-v2/aws" already imported (revive)
	awssdk "github.com/aws/aws-sdk-go-v2/aws"
	^
resources/providers/awslib/cached_region_selector.go:34:40: var-declaration: should drop = 0 from declaration of var currentRegionCacheTTL; it is the zero value (revive)
	currentRegionCacheTTL time.Duration = 0
	                                      ^
resources/providers/awslib/ec2/nacl.go:38:2: comment-spacings: no space between comment delimiter and comment text (revive)
	//arn:aws:ec2:region:account-id:network-acl/network-acl-id
	^
resources/providers/awslib/ecr/provider.go:32:2: comment-spacings: no space between comment delimiter and comment text (revive)
	/// When repoNames is nil, it will describe all the existing Repositories
	^
resources/fetching/fetchers/aws/elb_fetcher.go:124:1: receiver-naming: receiver name res should be consistent with previous receiver name r for ElbResource (revive)
func (res ElbResource) GetElasticCommonData() (map[string]any, error) {
	return map[string]interface{}{
		"cloud.service.name": "ELB",
	}, nil
}
flavors/benchmark/builder/data_provider.go:28:35: unused-parameter: parameter 'event' seems to be unused, consider removing or renaming it as _ (revive)
func (d dataProvider) EnrichEvent(event *beat.Event, _ fetching.ResourceMetadata) error { return nil }
                                  ^
flavors/benchmark/builder/id_provider.go:22:28: unused-parameter: parameter 'resourceType' seems to be unused, consider removing or renaming it as _ (revive)
func (p *idProvider) GetId(resourceType string, resourceId string) string {
                           ^
resources/fetching/preset/aws_org_preset_test.go:49:13: flag-parameter: parameter 'drain' seems to be a control flag, avoid control coupling (revive)
func subtest(t *testing.T, drain bool) {
            ^
```

### Related Issues
- Related https://github.com/elastic/cloudbeat/issues/1669

### Checklist
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -72,6 +71,60 @@ linters-settings:
...
+        disabled: true
+      - name: import-alias-naming
+        disabled: true
+      - name: import-shadowing
+        disabled: true
+      - name: indent-error-flow
+        disabled: true
+      - name: line-length-limit
+        disabled: true
+        arguments: [160]
```
Comment (oren-zohar) [inline .golangci.yaml:109]: why disable it if you set the line length?

File: .golangci.yaml [hunk 2]
```diff
@@ -72,6 +71,60 @@ linters-settings:
...
+      - name: redundant-import-alias
+        disabled: true
+      - name: unchecked-type-assertion
+        disabled: true
+      - name: unexported-naming
+        disabled: true
+      - name: unexported-return
+        disabled: true
+      - name: unhandled-error
+        disabled: true
```
Comment (oren-zohar) [inline .golangci.yaml:121]: why skipping it?

File: .golangci.yaml [hunk 3]
```diff
@@ -72,6 +71,60 @@ linters-settings:
...
+      - name: blank-imports
+        disabled: true
+      - name: cognitive-complexity
+        disabled: true
+      - name: confusing-naming
+        disabled: true
+      - name: confusing-results
+        disabled: true
+      - name: cyclomatic
+        disabled: true
```
Comment (oren-zohar) [inline .golangci.yaml:90]: I think this is a good way to check for readability. Why are we disabling it?

File: launcher/launcher_test.go [hunk 4]
```diff
@@ -289,6 +289,7 @@ func (s *LauncherTestSuite) TestWaitForUpdates() {
 	}
 
 	for _, tt := range testcases {
+		tt := tt
```
Comment (oren-zohar) [inline launcher/launcher_test.go:292]: ?

File: .golangci.yaml (hunk 2 above)
Comment (orestisfl) [reply .golangci.yaml:121]: Rules like this need larger changes so we can enable them one-by-one to make reviewing easier.

File: .golangci.yaml (hunk 3 above)
Comment (orestisfl) [reply .golangci.yaml:90]: Rules like this need larger changes so we can enable them one-by-one to make reviewing easier.

File: launcher/launcher_test.go (hunk 4 above)
Comment (orestisfl) [reply launcher/launcher_test.go:292]: Common golang idiom, the `tt` variable needs to be captured when used in goroutines.

See https://eli.thegreenplace.net/2019/go-internals-capturing-loop-variables-in-closures/

This is fixed in 1.22 but we are still not in that version.

**Review by oren-zohar — APPROVED** [review]

**Pull request merged**